core/src/main/java/greencity/repository/HabitRepo.java

```diff
@@ -72,7 +63,14 @@
      * @param end    last day of period
      * @return count of user habits that were marked during some period
      */
-    @Query("SELECT COUNT(h) FROM Habit h WHERE h.user.id = ?1 AND h.statusHabit = true AND h.createDate < ?2 AND h.id "
-        + "IN(SELECT hs.habit.id FROM HabitStatistic hs WHERE hs.createdOn BETWEEN ?2 and ?3)  ")
+    @Query("SELECT u.habits.size "
+        + "FROM User u "
+        + "WHERE u.id = ?1 "
+        + "AND u.habits IN(SELECT h FROM Habit h "
+        + "WHERE h.statusHabit = true "
```

Comment:
I was expecting you to replace all IN statements with JOINs.

Reply:
Sorry, little misunderstanding) Now I fixed all queries.